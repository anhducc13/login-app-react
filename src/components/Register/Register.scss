.Register {
  position: absolute  ;
  // top: 0;
  // bottom: 0;
  height: 100%;
  width: 100%;
  background-image: linear-gradient(to bottom, rgba(255,255,0,0.5), rgba(0,0,255,0.5)),
      url("../../assets/img/register.jpeg");
  background-repeat: no-repeat;
  background-size: cover;
  //background-position: center center;
  background-attachment: scroll;
  z-index: 1;
}

.form-box {
  margin-top: 30px;
  background-color: #ffffff;
  z-index: 10;
  padding: 10px 20px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.5);
  flex-direction: column;
}
