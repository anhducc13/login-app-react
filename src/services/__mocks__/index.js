// @flow
export { default as authServices } from './auth';