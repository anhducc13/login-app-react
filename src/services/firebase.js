import * as firebase from 'firebase';
import uuidv4 from 'uuid/v4';

const appFirebase = firebase.initializeApp({
    apiKey: "AIzaSyAcjbrVQyOOlY-SadQ8TWmldob2SjH5trM",
    authDomain: "ductt-217110.firebaseapp.com",
    databaseURL: "https://ductt-217110.firebaseio.com",
    storageBucket: "ductt-217110.appspot.com",
});

const uploadFileImage = (blob) => {
    const filename = Date.now().toString() + uuidv4();
    const ref = appFirebase.storage().ref(`avatar/${filename}`);
    return ref.put(blob).then(snapshot => snapshot.metadata.fullPath)
}

const getDownloadURL = (refName) => {
    const ref = appFirebase.storage().ref();
    return ref.child(refName).getDownloadURL();
}

const deleteFileImage = (refName) => {
    const ref = appFirebase.storage().ref();
    return ref.child(refName).delete();
}

export default {
    uploadFileImage,
    getDownloadURL,
    deleteFileImage,
};