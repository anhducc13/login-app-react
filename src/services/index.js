// @flow
export { default as userServices } from './user';
export { default as authServices } from './auth';
export { default as requestServices } from './requests';
export { default as firebaseServices } from './firebase';
