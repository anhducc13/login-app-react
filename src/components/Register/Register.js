import React, { useState } from 'react';
import {
  Form,
  Input,
  Button,
  Col,
  Row,
} from 'antd';
import { Link } from 'react-router-dom';
import './Register.scss';
import Title from 'antd/lib/typography/Title';
import { authServices } from 'services';
import { validatorHelpers } from 'helpers';
import openNotificationWithIcon from 'helpers/notification';

const { Password } = Input;


const Register = (props) => {
  const bodyLayout = {
    xs: { span: 22, offset: 1 },
    sm: { span: 16, offset: 4 },
    md: { span: 12, offset: 6 },
    lg: { span: 10, offset: 7 },
    xl: { span: 8, offset: 8 },
    xxl: { span: 6, offset: 9 },
  }
  const [loading, setLoading] = useState(false);

  const initState = {
    value: "",
    helper: ""
  }

  const [email, setEmail] = useState(initState);
  const [username, setUsername] = useState(initState);
  const [password, setPassword] = useState(initState);
  const [repeatPassword, setRepeatPassword] = useState(initState);

  const [errorText, setErrorText] = useState("")

  const validateUsername = (value) => {
    const helper = validatorHelpers.validateUsername(value);
    setUsername({ value, helper })
  }

  const validateEmail = (value) => {
    const helper = validatorHelpers.validateEmail(value);
    setEmail({ value, helper })
  }

  const validatePassword = (value) => {
    const helper = validatorHelpers.validatePassword(value);
    setPassword({ value, helper })
  }

  const validateRepeatPassword = (value) => {
    const helper = value === password.value ? "" : "Password repeat is not match";
    setRepeatPassword({ value, helper })
  }


  const handleSubmit = (e) => {
    e.preventDefault();

    validateEmail(email.value);
    validateUsername(username.value);
    validatePassword(password.value)
    validateRepeatPassword(repeatPassword.value)

    const canNotSubmit = email.value === "" || email.helper !== ""
      || username.value === "" || username.helper !== ""
      || password.value === "" || password.helper !== ""
      || repeatPassword.value === "" || repeatPassword.helper !== "";
    if (canNotSubmit) {
      return;
    }
    setLoading(true);
    const params = {
      "username": username.value,
      "email": email.value,
      "password": password.value,
    }
    authServices.registerUser(params)
      .then(res => {
        setLoading(false);
        const { data } = res;
        openNotificationWithIcon('success', 'Success',
          `Please verify account in email ${data.email} to login system`)
        props.history.push('/login');
      })
      .catch(err => {
        setLoading(false);
        if (err && err.response) {
          const { status, data } = err.response;
          if (status === 400) {
            setErrorText(data.message);
            return;
          }
          if (status === 404) {
            props.history.push('/404');
            return;
          }
          if (status > 400 && status < 500) {
            props.history.push('/403');
            return;
          }
        }
        props.history.push('/500');
      })
  }



  return (
    <div className="Register">
      <Col {...bodyLayout}>
        <Form className="form-box">
          <Title
            level={2}
            style={{
              padding: "5px",
              textAlign: "center"
            }}
          >
            Register
          </Title>
          <Title
            level={4}
            type="danger"
            style={{
              textAlign: "center"
            }}
          >
            {errorText}
          </Title>
          <Form.Item label="E-mail" help={email.helper} validateStatus={email.helper && "error"}>
            <Input size="large" name="email" onChange={e => validateEmail(e.target.value)} />
          </Form.Item>
          <Form.Item label="Username" help={username.helper} validateStatus={username.helper && "error"}>
            <Input size="large" name="username" onChange={e => validateUsername(e.target.value)} />
          </Form.Item>
          <Form.Item label="Password" help={password.helper} validateStatus={password.helper && "error"}>
            <Password size="large" name="password" onChange={e => validatePassword(e.target.value)} />
          </Form.Item>
          <Form.Item
            label="Repeat Password"
            help={repeatPassword.helper}
            validateStatus={repeatPassword.helper && "error"}
          >
            <Password size="large" name="repeat-password" onChange={e => validateRepeatPassword(e.target.value)} />
          </Form.Item>
          <Row>
            <Col style={{ textAlign: "center" }}>
              <Link to="/login">Have an account? Login</Link>
            </Col>
          </Row>
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              block
              loading={loading}
              onClick={handleSubmit}
              size="large"
              style={{
                marginTop: 10,
              }}
            >
              Register
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </div>
  );
};

export default Register;
