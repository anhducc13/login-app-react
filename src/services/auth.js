import { requestServices } from 'services';

const loginUser = (payload) => requestServices.customAxios.post('/auth/login',payload)

const registerUser = (payload) => requestServices.customAxios.post('/auth/register',payload)

const forgotPasswordUser = (payload) => requestServices.customAxios.post('/auth/forgotPassword',payload)

const updatePasswordUser = (payload) => requestServices.customAxios.post('/auth/updatePassword',payload)

const logoutUser = () => requestServices.customAxios.get('/auth/logout')

const currentUser = () => requestServices.customAxios.get('/auth/currentUser')


  export default {
    loginUser,
    currentUser,
    registerUser,
    forgotPasswordUser,
    logoutUser,
    updatePasswordUser
  };