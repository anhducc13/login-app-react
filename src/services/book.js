// @flow
import { requestServices } from 'services';

const fetchBooks = (args) => requestServices.customAxios.get('/book/list/', { params: args })
  .then(res => res.data.data);

const getAllCategory = () => requestServices.customAxios.get(`/book/allCategory`)
  .then(res => res.data.data);

const fetchCategoryActions = (args) => requestServices.customAxios.get(`/category/activity`, { params: args })
  .then(res => res.data.data);

const addBook = (params) => requestServices.customAxios.post('/book/', params, {headers: {
  'Content-Type': 'application/json',
}})
  .then(res => res.data.data);

const editCategory = (id, params) => requestServices.customAxios.put(`/category/${id}`, params)
  .then(res => res.data.data);

const deleteCategory = (id) => requestServices.customAxios.delete(`/category/${id}`)
  .then(res => res.data.data);

export default {
  fetchBooks,
  getAllCategory,
  addBook,
  editCategory,
  deleteCategory,
  fetchCategoryActions,
};
