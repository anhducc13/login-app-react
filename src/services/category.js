// @flow
import { requestServices } from 'services';

const fetchCategories = (args) => requestServices.customAxios.get('/category/list/', { params: args })
  .then(res => res.data.data);

const fetchCategory = (id) => requestServices.customAxios.get(`/category/${id}`)
  .then(res => res.data.data);

const addCategory = (params) => requestServices.customAxios.post('/category/', params)
  .then(res => res.data.data);

const editCategory = (id, params) => requestServices.customAxios.put(`/category/${id}`, params)
  .then(res => res.data.data);

const deleteCategory = (id) => requestServices.customAxios.delete(`/category/${id}`)
  .then(res => res.data.data);

export default {
  fetchCategories,
  fetchCategory,
  addCategory,
  editCategory,
  deleteCategory,
};
