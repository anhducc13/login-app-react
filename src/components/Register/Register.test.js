import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { authServices } from 'services';
import Register from './Register';


describe('Register Page Render', () => {
  it('should render correctly', () => {
    const component = shallow(<Register />);
    expect(component).toMatchSnapshot();
  });
});

describe('Register form validate', () => {
  it('validate test case 1', () => {
    const component = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    component.find('input[name="username"]').simulate('change', { target: { value: "" } });
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Username not empty")
    component.unmount()
  })

  it('validate test case 2', () => {
    const component = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "ab" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "anhducc14" } });
    component.find('input[name="password"]').simulate('change', { target: { value: "a" } });
    component.find('input[name="repeat-password"]').simulate('change', { target: { value: "a" } });
    expect(component.find('div.ant-form-explain').length)
      .toBe(2)
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Email is invalid format")
    component.unmount()
  })

  it('validate test case 3', () => {
    const component = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "trantienduc10@gmail.com" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "anhducc13" } });
    component.find('input[name="password"]').simulate('change', { target: { value: "Anhducc13" } });
    component.find('input[name="repeat-password"]').simulate('change', { target: { value: "Anhducc13" } });
    expect(component.find('div.ant-form-explain').length)
      .toBe(0)
    component.unmount();
  })
});

describe('Register Page Submit Form', () => {
  let registerSpy;

  beforeAll(() => {
    registerSpy = jest.fn();
    registerSpy = jest.spyOn(authServices, 'registerUser');
  });

  it('Submit register will not call registerUser function after fail validate', () => {
    const component = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    component.find('input[name="username"]').simulate('change', { target: { value: "anhducc" } });
    component.find('input[name="password"]').simulate('change', { target: { value: "Anhducc14" } });
    component.find('button[type="submit"]').simulate('click');
    expect(registerSpy)
      .not.toHaveBeenCalled()
    component.unmount()
  })

  it('Submit login will call loginUser function after success validate', () => {
    const component = mount(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "trantienduc10@gmail.com" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "anhducc14" } });
    component.find('input[name="password"]').simulate('change', { target: { value: "Anhducc14" } });
    component.find('input[name="repeat-password"]').simulate('change', { target: { value: "Anhducc14" } });
    component.find('button[type="submit"]').simulate('click');
    expect(registerSpy)
      .toHaveBeenCalled()
    expect(registerSpy)
      .toHaveBeenCalledWith({
        "email": "trantienduc10@gmail.com",
        "username": "anhducc14",
        "password": "Anhducc14"
      })
    component.unmount()
  })

  afterAll(() => {
    jest.restoreAllMocks();
  });
})