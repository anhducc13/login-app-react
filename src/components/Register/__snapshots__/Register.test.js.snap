// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Render should render correctly 1`] = `
<div
  className="Register"
>
  <Col
    lg={
      Object {
        "offset": 7,
        "span": 10,
      }
    }
    md={
      Object {
        "offset": 6,
        "span": 12,
      }
    }
    sm={
      Object {
        "offset": 4,
        "span": 16,
      }
    }
    xl={
      Object {
        "offset": 8,
        "span": 8,
      }
    }
    xs={
      Object {
        "offset": 1,
        "span": 22,
      }
    }
    xxl={
      Object {
        "offset": 9,
        "span": 6,
      }
    }
  >
    <Form
      className="form-box"
      colon={true}
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <Title
        level={2}
        style={
          Object {
            "padding": "5px",
            "textAlign": "center",
          }
        }
      >
        Register
      </Title>
      <Title
        level={4}
        style={
          Object {
            "textAlign": "center",
          }
        }
        type="danger"
      />
      <FormItem
        hasFeedback={false}
        help=""
        label="E-mail"
        validateStatus=""
      >
        <Input
          disabled={false}
          name="email"
          onChange={[Function]}
          size="large"
          type="text"
        />
      </FormItem>
      <FormItem
        hasFeedback={false}
        help=""
        label="Username"
        validateStatus=""
      >
        <Input
          disabled={false}
          name="username"
          onChange={[Function]}
          size="large"
          type="text"
        />
      </FormItem>
      <FormItem
        hasFeedback={false}
        help=""
        label="Password"
        validateStatus=""
      >
        <Password
          action="click"
          inputPrefixCls="ant-input"
          name="password"
          onChange={[Function]}
          prefixCls="ant-input-password"
          size="large"
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        hasFeedback={false}
        help=""
        label="Repeat Password"
        validateStatus=""
      >
        <Password
          action="click"
          inputPrefixCls="ant-input"
          name="repeat-password"
          onChange={[Function]}
          prefixCls="ant-input-password"
          size="large"
          visibilityToggle={true}
        />
      </FormItem>
      <Row
        gutter={0}
      >
        <Col
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <Link
            to="/login"
          >
            Have an account? Login
          </Link>
        </Col>
      </Row>
      <FormItem
        hasFeedback={false}
      >
        <Button
          block={true}
          ghost={false}
          htmlType="submit"
          loading={false}
          onClick={[Function]}
          size="large"
          style={
            Object {
              "marginTop": 10,
            }
          }
          type="primary"
        >
          Register
        </Button>
      </FormItem>
    </Form>
  </Col>
</div>
`;
